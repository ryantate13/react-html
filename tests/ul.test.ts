import {ul} from '../src';

describe('ul', () => {
    it('renders with children', () => {
        expect(ul({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(ul({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(ul()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(ul({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(ul({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(ul({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(ul({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(ul({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(ul({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(ul({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(ul({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(ul({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(ul({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(ul({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(ul({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(ul({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(ul({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(ul({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(ul({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(ul({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(ul({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(ul({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(ul({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(ul({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(ul({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(ul({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(ul({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(ul({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(ul({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(ul({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(ul({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(ul({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(ul({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(ul({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(ul({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(ul({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(ul({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(ul({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(ul({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(ul({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(ul({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(ul({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(ul({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(ul({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(ul({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(ul({translate: "no"})).toMatchSnapshot();
    });
});