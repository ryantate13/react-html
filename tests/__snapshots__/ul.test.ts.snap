// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ul renders with accessKey="a,b,c,d" 1`] = `
<ul
  accessKey="a,b,c,d"
/>
`;

exports[`ul renders with autoCapitalize="characters" 1`] = `
<ul
  autoCapitalize="characters"
/>
`;

exports[`ul renders with autoCapitalize="none" 1`] = `
<ul
  autoCapitalize="none"
/>
`;

exports[`ul renders with autoCapitalize="off" 1`] = `
<ul
  autoCapitalize="off"
/>
`;

exports[`ul renders with autoCapitalize="on" 1`] = `
<ul
  autoCapitalize="on"
/>
`;

exports[`ul renders with autoCapitalize="sentences" 1`] = `
<ul
  autoCapitalize="sentences"
/>
`;

exports[`ul renders with autoCapitalize="words" 1`] = `
<ul
  autoCapitalize="words"
/>
`;

exports[`ul renders with children 1`] = `
<ul>
  TEST
</ul>
`;

exports[`ul renders with className="foo bar baz" 1`] = `
<ul
  className="foo bar baz"
/>
`;

exports[`ul renders with contentEditable="false" 1`] = `
<ul
  contentEditable="false"
/>
`;

exports[`ul renders with contentEditable="true" 1`] = `
<ul
  contentEditable="true"
/>
`;

exports[`ul renders with dir="auto" 1`] = `
<ul
  dir="auto"
/>
`;

exports[`ul renders with dir="ltr" 1`] = `
<ul
  dir="ltr"
/>
`;

exports[`ul renders with dir="rtl" 1`] = `
<ul
  dir="rtl"
/>
`;

exports[`ul renders with draggable="false" 1`] = `
<ul
  draggable="false"
/>
`;

exports[`ul renders with draggable="true" 1`] = `
<ul
  draggable="true"
/>
`;

exports[`ul renders with empty attributes 1`] = `<ul />`;

exports[`ul renders with hidden="false" 1`] = `
<ul
  hidden={false}
/>
`;

exports[`ul renders with hidden="true" 1`] = `
<ul
  hidden={true}
/>
`;

exports[`ul renders with id="TEST" 1`] = `
<ul
  id="TEST"
/>
`;

exports[`ul renders with inputMode="decimal" 1`] = `
<ul
  inputMode="decimal"
/>
`;

exports[`ul renders with inputMode="email" 1`] = `
<ul
  inputMode="email"
/>
`;

exports[`ul renders with inputMode="none" 1`] = `
<ul
  inputMode="none"
/>
`;

exports[`ul renders with inputMode="numeric" 1`] = `
<ul
  inputMode="numeric"
/>
`;

exports[`ul renders with inputMode="search" 1`] = `
<ul
  inputMode="search"
/>
`;

exports[`ul renders with inputMode="tel" 1`] = `
<ul
  inputMode="tel"
/>
`;

exports[`ul renders with inputMode="text" 1`] = `
<ul
  inputMode="text"
/>
`;

exports[`ul renders with inputMode="url" 1`] = `
<ul
  inputMode="url"
/>
`;

exports[`ul renders with is="foo" 1`] = `
<ul
  is="foo"
/>
`;

exports[`ul renders with itemID="https://google.com/" 1`] = `
<ul
  itemID="https://google.com/"
/>
`;

exports[`ul renders with itemProp="https://google.com/" 1`] = `
<ul
  itemProp="https://google.com/"
/>
`;

exports[`ul renders with itemRef="foo bar" 1`] = `
<ul
  itemRef="foo bar"
/>
`;

exports[`ul renders with itemScope="false" 1`] = `
<ul
  itemScope={false}
/>
`;

exports[`ul renders with itemScope="true" 1`] = `
<ul
  itemScope={true}
/>
`;

exports[`ul renders with itemType="https://google.com/" 1`] = `
<ul
  itemType="https://google.com/"
/>
`;

exports[`ul renders with lang="" 1`] = `
<ul
  lang=""
/>
`;

exports[`ul renders with lang="es-419" 1`] = `
<ul
  lang="es-419"
/>
`;

exports[`ul renders with slot="TEST" 1`] = `
<ul
  slot="TEST"
/>
`;

exports[`ul renders with spellCheck="false" 1`] = `
<ul
  spellCheck="false"
/>
`;

exports[`ul renders with spellCheck="true" 1`] = `
<ul
  spellCheck="true"
/>
`;

exports[`ul renders with style="[object Object]" 1`] = `
<ul
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`ul renders with tabIndex="1" 1`] = `
<ul
  tabIndex={1}
/>
`;

exports[`ul renders with title="TEST" 1`] = `
<ul
  title="TEST"
/>
`;

exports[`ul renders with translate="no" 1`] = `
<ul
  translate="no"
/>
`;

exports[`ul renders with translate="yes" 1`] = `
<ul
  translate="yes"
/>
`;

exports[`ul renders without attributes 1`] = `<ul />`;
